src/Compilers/VisualBasic/Portable/Binding/LookupResult.vb

```diff
@@ -722,6 +728,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             MergeAmbiguous(other, s_ambiguousInTypeError)
         End Sub
 
+        Private Function AreEquivalentEnumConstants(symbol1 As Symbol, symbol2 As Symbol) As Boolean
+            If symbol1.Kind <> SymbolKind.Field OrElse symbol2.Kind <> SymbolKind.Field OrElse symbol1.ContainingType.TypeKind <> TypeKind.Enum Then
```

Comment:
No. This is called from a method that gathers members from a type. In fact both are members of the same type. I will add such an assertion.